cache/src/main/java/io/envoyproxy/controlplane/cache/SnapshotCache.java

```diff
@@ -10,4 +10,12 @@
    * @param snapshot a versioned collection of node config data
    */
   void setSnapshot(T group, Snapshot snapshot);
+
+  /**
+   * Returns the most recently set {@link Snapshot} for the given node group.
+   *
+   * @param group group identifier
+   * @return latest snapshot
+   */
+  Snapshot getSnapshot(T group);
```

Comment:
nit: could you move `getSnapshot` above `setSnapshot` to keep them in alphabetical order? ditto on the implementation.